eogrow/pipelines/export_maps.py

```diff
@@ -235,51 +212,142 @@ def _prepare_files(self, geotiff_paths: List[str], output_file: str) -> Tuple[FS
             temp_geotiff_paths,
             workers=None,
             multiprocess=False,
-            desc="Making local copies",
         )
-        temp_map_path = fs.path.basename(output_file)
-        return temp_fs, temp_geotiff_paths, temp_map_path
+        return temp_fs, temp_geotiff_paths
+
+    def _split_patches_temporally(
+        self, filesystem: FS, output_folder: str, geotiff_paths: List[str], some_eopatch: str, crs: CRS
+    ) -> Dict[Optional[dt.datetime], List[str]]:
+        """Splits input patches into multiple tiffs according to the temporal dimension.
+
+        Here the jobs are grouped together per-inputfile, in later parts jobs are grouped per-time.
+        Due to this jobs are created per-input, but we also take note of per-time grouping, which the function returns.
+        """
+        LOGGER.info("Splitting TIFF files temporally.")
+        num_bands, timestamp = self._load_bands_and_timestamp(some_eopatch)  # assume eopatches share these
+
+        def get_extraction_path(input_path: str, time: dt.datetime) -> str:
+            """Ensures that after extraction the files have a time-suffix."""
+            tiff_name = _strip_tiff_extension(fs.path.basename(input_path))
+            return fs.path.join(output_folder, get_tiff_name(self.map_name, tiff_name, crs, time))
+
+        geotiffs_per_time: Dict[Optional[dt.datetime], List[str]] = defaultdict(list)
+        grouped_split_jobs = []  # these are grouped per input file to avoid IO blocking of processes
+        for tiff_path in geotiff_paths:
+            tiff_sys_path = filesystem.getsyspath(tiff_path)
 
-    def _load_timestamp(self, eopatch_name: str) -> List[dt.datetime]:
+            jobs = []
+            for i, time in enumerate(timestamp):
+                extraction_path = get_extraction_path(tiff_path, time)
+                bands = range(i * num_bands, (i + 1) * num_bands)
+                jobs.append(SplitTiffsJob(tiff_sys_path, bands, filesystem.getsyspath(extraction_path)))
+
+                geotiffs_per_time[time].append(extraction_path)
+
+            grouped_split_jobs.append(jobs)
+
+        parallelize(self._execute_split_jobs, grouped_split_jobs, workers=None)
+        parallelize(filesystem.remove, geotiff_paths, workers=None, multiprocess=False)
+
+        return geotiffs_per_time
+
+    def _load_bands_and_timestamp(self, eopatch_name: str) -> Tuple[int, List[dt.datetime]]:
```

Comment:
i decided to pull the number of bands from the eopatch since we need to load the timestamp anyway. Also this is done for a single EOPatch, so i'm not all that worried about time complexity here. I don't want to use `gdalinfo` since subprocess calls are a nice bullet trying to get into my knee, but the previous rasterio approach would work just fine. I just don't see much reason for it tbh.

Moving this before the CRS loop would work, but then i'd need to thread them as further parameters through the workflow, resulting in even longer function calls.

Reply:
name was changed

Reply:
resolving since the `gdalinfo` debate moved a few threads down